desktop/app.js

```diff
@@ -164,20 +156,37 @@ module.exports = function main() {
     mainWindow.once('ready-to-show', mainWindow.show);
   };
 
-  const gotTheLock = app.requestSingleInstanceLock();
+  app.on('will-finish-launching', function () {
+    setTimeout(updater.ping.bind(updater), config.updater.delay);
+    app.on('open-url', function (event, url) {
+      event.preventDefault();
+      mainWindow.webContents.send('wpLogin', url);
+    });
+  });
 
-  app.on('second-instance', () => {
-    // Someone tried to run a second instance, we should focus our window.
-    if (mainWindow) {
-      if (mainWindow.isMinimized()) {
-        mainWindow.restore();
+  const gotTheLock = app.requestSingleInstanceLock();
+  if (gotTheLock) {
+    app.on('second-instance', (e, argv) => {
+      // Someone tried to run a second instance, we should focus our window.
+      // Protocol handler for win32
+      // argv: An array of the second instance’s (command line / deep linked) arguments
+      if (process.platform !== 'darwin') {
+        // Keep only command line / deep linked arguments
+        const url =
+          (argv.slice(-1) && argv.slice(-1).length && argv.slice(-1)[0]) || '';
```

Comment:
can we show in a comment what these might be that we expect? it now looks like we're trying to get the last-passed arg from the CLI but it's not lear what that argument would or should be

(The comment I'm leaving in this snippet might be complete garbage but that's just because I'm trying to understand what's happening but don't know).
```js
app.on( 'second-instance', (event, argv) => {
	// when we register the `simplenote://` protocol in Windows and Linux they handle those
	// links by spawning a new invocation of Electron and pass the URL as the last parameter
	// in this we case we expect an instance to be already open and so we'll intercept that event
	// here and translate the URL into an action for the app to handle
	//
	// e.g. /some/path/to/simplenote.appx http://localhost:4000
	//
	// in macOS it doesn't send the URL so we'll assume this is for WPCC login but eventually
	// we'll have to differentiate to support deep-linking to individual notes if we add that

	const isMac = process.platform === 'darwin';
	const loginUrl = isMac ? url : argv[ argv.length - 1]

	mainWindow.webContents.send('wpLogin', url);
} );
```

after writing this comment I believe I understand the problem better but it begs a few questions:
 - am I correct in what I described in the comment?
 - what is the role of the URL? is it passed back from the WPCC auth server managed in the OAuth client application? (the `redirect_url`)?
 - why is macOS different?
 - should we verify that the URL is well-formed according to our expectations? should we differentiate what we do based on the URL?

Reply:
macOS uses the open-url event. macOS does allow for multiple instances. At least it is not supposed to. There were some bugs saying that this event was fired on macOS, thus the check.

Reply:
We only want to run the code if it is not macOS

Reply:
> macOS uses the open-url event

this would be good to include in the comment. I can easily see someone spending time trying to understand why something is different on Windows/macOS where that little sentence could save them (or us) time.

> We only want to run the code if it is not macOS

ah. my code in PR comments is usually wrong at least in some way.

```js
if (process.platform === 'darwin') {
	// macOS communicates deep-linking via the `open-url` event (see above)
	// but we might still end up with this message so ignore it if we do
	return;
}
```